Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+                                                           'createUserMapping', 'removeUserDirectory']
+
+
+  ADMIN_METHODS = FileCatalogClientBase.ADMIN_METHODS + ['getUserDirectory', 'createUserDirectory',
+                                                         'createUserMapping', 'removeUserDirectory']
+
+  def __init__(self, **options):
+    """ Constructor
+    """
+    self.convertUnicode = True
+    proxyInfo = {'OK': False}
+    if os.getenv('RUCIO_AUTH_TYPE') == 'x509_proxy' and not os.getenv('X509_USER_PROXY'):
+      proxyInfo = getProxyInfo(disableVOMS=True)
+      if proxyInfo['OK']:
+        os.environ['X509_USER_PROXY'] = proxyInfo['Value']['path']
+        sLog.debug('X509_USER_PROXY not defined. Using %s' % proxyInfo['Value']['path'])
+    try:
+      self._client = Client()
+      self.account = self._client.account
+    except (CannotAuthenticate, MissingClientParameter):
+      if os.getenv('RUCIO_AUTH_TYPE') == 'x509_proxy':
+        if not proxyInfo['OK']:
+          proxyInfo = getProxyInfo(disableVOMS=True)
+        if proxyInfo['OK']:
+          dn = proxyInfo['Value']['identity']
+          username = proxyInfo['Value']['username']
+          self.account = username
+          sLog.debug('Switching to account %s mapped to proxy %s' %(username, dn))
+
+    try:
+      self._client = Client(account=self.account)
+      self.scopes = self._client.list_scopes()
+    except Exception as err:
+        sLog.error('Cannot instantiate RucioFileCatalog interface', 'error : %s' % repr(err))
+
+  @property
+  def client(self):
+    """ Check if the session to the server is still active and return an instance of RucioClient
+    """
+    try:
+      self._client.ping()
+      return self._client
+    except:
+      self._client = Client(account=self.account)
+      self.scopes = self._client.list_scopes()
+      return self._client
+
+
+  def __getDidsFromLfn(self, lfn):
+    """ Helper function to convert LFNs into DIDs
+    """
+    if lfn.find(':') > -1:
+      scope, name = lfn.split(':')
+      return {'scope': scope, 'name': name}
+    else:
+      scope = get_scope(lfn, scopes=self.scopes)
+    return {'scope': scope, 'name': lfn}
+
+
+  @checkCatalogArguments
+  def getCompatibleMetadata(self, queryDict, path, credDict):
+    """ Get distinct metadata values compatible with the given already defined metadata
+    """
+    if path != '/':
+      result = self.exists(path)
+      if not result['OK']:
+        return result
+      if not result['Value']:
+        return S_ERROR('Path not found: %s' % path)
+    result = S_OK({})
+    return result
+
+
+  @checkCatalogArguments
+  def getReplicas(self, lfns, allStatus = False):
+    """ Returns replicas for an LFN or list of LFNs
+    """
+    result = {'OK': True, 'Value': {'Successful': {}, 'Failed': {}}}
+    lfnChunks = breakListIntoChunks( lfns, 1000 )
+
+    for lfnList in lfnChunks:
```

Comment:
No need for the `lfnChunks` variable (especially as I think this produced a generator so reusing `lfnChunks` will give wrong results).